src/DataModel/IApiDocumentation.ts

```diff
@@ -0,0 +1,9 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import {IData} from "./IData";
+
+export interface IApiDocumentation extends IMetadataDescription
+{
+    entryPoint: string;
+
+    getEntryPoint(): Promise<IData>;
```

Comment:
That's why I'm torn apart - now I somehow tend to stick to that latter approach where client does all the stuff - I'm just worried that the client would be overloaded with logic.
The thing is that code equivalent following that approach:
```javascript
var api = client.get("http://example.com");
var apiDocumentation = client.get(api.getApiDocumentationUrl());
var entryPoint = client.get(apiDocumentation.entryPoint);
var operation = entryPoint
    .filter(op => op.method == "POST" && op.expects == "Event")[0];
client.invoke(operation, event);
```
looks a bit messy